gateleen-security/src/main/java/org/swisspush/gateleen/security/authorization/RoleAuthorizer.java

```diff
@@ -0,0 +1,217 @@
+
+import java.util.*;
+import java.util.Map.Entry;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+public class RoleAuthorizer implements ConfigurationResource {
+
+    private String aclRoot;
+    private String aclKey = "acls";
+
+    private String adminRole = "admin";
+    private String anonymousRole = "everyone";
+    private String deviceHeader = "x-rp-deviceid";
+    private String userHeader = "x-rp-usr";
+
+    private AclFactory aclFactory;
+    private RoleMapper roleMapper;
+
+    private PatternHolder aclUriPattern;
+
+    private ResourceStorage storage;
+    private RoleExtractor roleExtractor;
+    private Map<PatternHolder, Map<String, Set<String>>> initialGrantedRoles;
+
+    // URI -> Method -> Roles
+    private Map<PatternHolder, Map<String, Set<String>>> grantedRoles = new HashMap<>();
+
+    public static final Logger log = LoggerFactory.getLogger(RoleAuthorizer.class);
+
+    public RoleAuthorizer( final ResourceStorage storage, String securityRoot, String rolePattern,final RoleMapper roleMapper) {
+        this.storage = storage;
+        this.roleMapper = roleMapper;
+        this.aclRoot = securityRoot + aclKey + "/";
+        this.aclUriPattern = new PatternHolder(Pattern.compile("^" + aclRoot + "(?<role>.+)$"));
+        this.roleExtractor = new RoleExtractor(rolePattern);
+        this.aclFactory = new AclFactory();
+
+        initialGrantedRoles = new HashMap<>();
+        initialGrantedRoles.put(aclUriPattern, new HashMap<>());
+        initialGrantedRoles.get(aclUriPattern).put("PUT", new HashSet<>());
+        initialGrantedRoles.get(aclUriPattern).put("GET", new HashSet<>());
+        initialGrantedRoles.get(aclUriPattern).put("DELETE", new HashSet<>());
+        initialGrantedRoles.get(aclUriPattern).get("PUT").add(adminRole);
+        initialGrantedRoles.get(aclUriPattern).get("GET").add(adminRole);
+        initialGrantedRoles.get(aclUriPattern).get("DELETE").add(adminRole);
+
+        configUpdate();
+    }
+
+    public void handleIsAuthorized(final HttpServerRequest request, Future<Boolean> future) {
+        if (!isAuthorized(request)) {
+            ResponseStatusCodeLogUtil.info(request, StatusCode.FORBIDDEN, RoleAuthorizer.class);
+            request.response().setStatusCode(StatusCode.FORBIDDEN.getStatusCode());
+            request.response().setStatusMessage(StatusCode.FORBIDDEN.getStatusMessage());
+            request.response().end(StatusCode.FORBIDDEN.getStatusMessage());
+            future.complete(Boolean.FALSE);
+        }
+    }
+
+    @Override
+    public void checkConfigResource(Buffer buffer) throws ValidationException {
+        aclFactory.parseAcl(buffer);
+    }
+
+    @Override
+    public void configUpdate() {
+        storage.get(aclRoot, buffer -> {
+            if (buffer != null) {
+                grantedRoles = new HashMap<>();
+                for (Object roleObject : new JsonObject(buffer.toString("UTF-8")).getJsonArray(aclKey)) {
```

Comment:
No need to convert to a String - you can directly use `buffer` to convert to JsonObject